pkg/build/api/helpers.go

```diff
@@ -59,3 +60,21 @@ func hasBuildConfigAnnotation(build Build, annotationName, annotationValue strin
 	value, ok := build.Annotations[annotationName]
 	return ok && value == annotationValue
 }
+
+// FindTriggerPolicy retrieves the BuildTrigger of a given type from a build configuration.
+// Returns an error if no match trigger is found.
+func FindTriggerPolicy(triggerType BuildTriggerType, config *BuildConfig) (buildTriggers []BuildTriggerPolicy, err error) {
+	err = fmt.Errorf("buildconfig trigger type not found: %v", triggerType)
+	for _, specTrigger := range config.Spec.Triggers {
+		if specTrigger.Type == triggerType {
+			buildTriggers = append(buildTriggers, specTrigger)
+			err = nil
+		}
+	}
+	return buildTriggers, err
```

Comment:
Seems weird to return an error for this.  i would expect the caller to check the array length and if it's zero, they know it wasn't found.  errors should be used for issues that are harder to anticipate imho.